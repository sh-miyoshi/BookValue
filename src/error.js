import React, { Component } from 'react';
import { Text } from 'react-native';

export class Error extends Component {
  constructor(props) {
    super(props);
  }

  render() {
    return (
      <Text style={[styles.textStyle, styles.errorTextStyle]}>
        エラー発生！{"\n"}
        {this.props.message}
      </Text>
    )
  }
}

const styles = {
  textStyle: {
    fontSize: 20,
    marginBottom: 20,
    textAlign: 'center',
    marginHorizontal: 15
  },
  errorTextStyle: {
    color: "rgba(255,0,0,1)",
  },
}