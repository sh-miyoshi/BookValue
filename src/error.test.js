import React from 'react'
import { Error } from './error'
import renderer from 'react-test-renderer'

it('renders correctly', () => {
  const tree = renderer
    .create(<Error message="test" />)
    .toJSON();
  expect(tree).toMatchSnapshot();
})