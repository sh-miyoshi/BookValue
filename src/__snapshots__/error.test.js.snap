// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders correctly 1`] = `
<Text
  accessible={true}
  allowFontScaling={true}
  ellipsizeMode="tail"
  style={
    Array [
      Object {
        "fontSize": 20,
        "marginBottom": 20,
        "marginHorizontal": 15,
        "textAlign": "center",
      },
      Object {
        "color": "rgba(255,0,0,1)",
      },
    ]
  }
>
  エラー発生！
  

  test
</Text>
`;
